src/main/java/graphql/execution/FieldCollectorParameters.java

```diff
@@ -70,12 +69,12 @@ public Builder objectType(GraphQLObjectType objectType) {
         }
 
         public Builder fragments(Map<String, FragmentDefinition> fragmentsByName) {
-            this.fragmentsByName.putAll(fragmentsByName);
+            this.fragmentsByName = fragmentsByName;
             return this;
         }
 
         public Builder variables(Map<String, Object> variables) {
-            this.variables.putAll(variables);
+            this.variables = variables;
             return this;
```

Comment:
This is a tricky one - generally we try for more immutable classes and hence by "copying" we gain some control in mutable Java land

However as this class is repeatedly called, we probably need to change this general pattern if we see evidence that it costs us.

Reply:
yes, I'm not sure too. I did that because the class is marked as @Internal (so we may have more freedom) but I know that it is against the pattern mutable/builder -> immutable/built object. 

It looks like it is always called with **ExecutionContext** data, that is unmodifiable.